test/download_helper.js

```diff
@@ -0,0 +1,257 @@
+  var elem = document.getElementById('test-nodes');
+  if(elem){
+    elem.parentNode.removeChild(elem);
+  }
+}
+
+describe("Download Helper", () => {
+  let dh;
+  let sandbox;
+
+  before(async () => {
+    sandbox = sinon.createSandbox();
+
+    dh = new DownloadHelper();
+
+    // Prevent Logger output during tests
+    sandbox.stub(dh, "log");
+  });
+
+  afterEach(() => {
+     // Preemptive removal of any test-nodes
+     cleanupTestNodes();
+     sandbox.restore();
+   });
+
+  describe("generateDownloadList()", () => {
+    it('should create a string of curl calls with urls from a.item-button', async () => {
+      createDomNodes(`
+        <span id="testId" class="download-title">
+          <a class="item-button" href="url1"></button>
+          <a class="item-button" href="url2"></button>
+        </span>
+      `)
+      const curlCommand = "curl -OJ url1 \\ &\ncurl -OJ url2"
+      const downloadList = DownloadHelper.generateDownloadList();
+      expect(downloadList == curlCommand).to.be.true;
+    });
+  });
+
+  describe("createButton()", () => {
+    it('should add a button to the end of div.download-titles', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles"></div>
+      `);
+
+      dh.createButton();
+      const button = document.querySelector(".downloadall")
+
+      expect(button != null).to.be.true;
+    });
+
+    it('should not replace existing button at the end of div.download-titles', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles">
+          <button class="downloadall fake" id="someId"></button>
+        </div>
+      `)
+
+      {
+        const button = document.querySelector(".downloadall")
+        const idAttribute = button.getAttribute("id");
+        expect(idAttribute == "someId").to.be.true;
+      }
+
+      dh.createButton();
+      
+      {
+        const button = document.querySelectorAll(".downloadall")
+        assert(button.length == 1);
+        const idAttribute = button[0].getAttribute("id");
+        expect(idAttribute == "someId").to.be.true;
+      }
+    });
+
+    it('should create specific properties for the button', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles"></div>
+      `)
+      
+      dh.createButton();
+
+      const buttonClassName = "downloadall"
+      expect(dh.button.className == buttonClassName).to.be.true;
+
+      const buttonTitleExpected = "Generates a file for automating downloads using 'curl'"
+      expect(dh.button.title == buttonTitleExpected).to.be.true;
+
+      expect(dh.button.disabled).to.be.true;
+
+      expect(dh.button.textContent == "preparing download").to.be.true;
+    });
+  });
+
+  describe("enableButton()", () => {
+    it('button should call functions on click', async () => {
+      DownloadHelper.dateString = sinon.fake.returns("dateString");
+      DownloadHelper.generateDownloadList = sinon.fake.returns("downloadList");
```

Comment:
The usage of the uninstantiated `DownloadHelper` class makes me a little uneasy here. Clearly this works (to my surprise) because the CI tests have passed, so that's cool. I'm worried that mocking these methods here at this point is too late, because `dh` is already defined in a `beforeEach` higher up. 

This might not be an issue for methods that don't manipulate class state, though I wonder if that's something we'll have to watch out for in the future. I'd suggest making sure that you're giving those methods really unique names to be sure that the spies are getting called with the values you've set directly in this test, and not leftovers from default or some other test.

Reply:
@emcniece these are "static" methods of the class so they operate without an instantiated class (in fact, they have no access to instantiated objects or their data).